src/ConnectedMode.UnitTests/ServerSentEvents/ServerSentEventPumpTests.cs

```diff
@@ -0,0 +1,116 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using SonarLint.VisualStudio.ConnectedMode.ServerSentEvents;
+using SonarLint.VisualStudio.Core.ServerSentEvents.Issues;
+using SonarLint.VisualStudio.Core.ServerSentEvents.TaintVulnerabilities;
+using SonarLint.VisualStudio.Integration.UnitTests;
+using SonarQube.Client.Models.ServerSentEvents;
+using SonarQube.Client.Models.ServerSentEvents.ClientContract;
+
+namespace SonarLint.VisualStudio.ConnectedMode.UnitTests.ServerSentEvents
+{
+    [TestClass]
+    public class ServerSentEventPumpTests
+    {
+        private Mock<IServerSentEventsSession> serverSentEventsSessionMock;
+        private Mock<IIssueChangedServerEventSourcePublisher> issueChangedPublisherMock;
+        private Mock<ITaintServerEventSourcePublisher> taintPublisherMock;
+        private ServerSentEventPump testSubject;
+
+        [TestInitialize]
+        public void SetUp()
+        {
+            serverSentEventsSessionMock = new Mock<IServerSentEventsSession>();
+            issueChangedPublisherMock = new Mock<IIssueChangedServerEventSourcePublisher>();
+            taintPublisherMock = new Mock<ITaintServerEventSourcePublisher>();
+
+            testSubject = new ServerSentEventPump(issueChangedPublisherMock.Object, taintPublisherMock.Object);
+        }
+
+        [TestMethod]
+        public void Dispose_DisposesPublishers()
+        {
+            testSubject.Dispose();
+
+            issueChangedPublisherMock.Verify(ch => ch.Dispose(), Times.Once);
+            taintPublisherMock.Verify(ch => ch.Dispose(), Times.Once);
+        }
+
+        [TestMethod]
+        public void MefCtor_CheckIsExported()
```

Comment:
Conventions: we try to keep the MEF test first in the class, with Dispose being last.